File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        
        Arrays.sort(participant);
        Arrays.sort(completion);

        for(int i = 0; i < participant.length; i++){
            if(i + 1 == participant.length) {
                answer = participant[i];
                break;
            }
            if(!participant[i].equals(completion[i])){
                answer = participant[i];
                break;
            }

        }


        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
from collections import Counter
def solution(participant, completion):
    answer = Counter(participant) - Counter(completion) # 키(이름) 기준 값(이름이 나온 횟수)을 서로 뺌 (중복제거)
    
    return next(iter(answer))                           # 딕셔너리에 마지막으로 남은(첫번째) 키만 뽑음